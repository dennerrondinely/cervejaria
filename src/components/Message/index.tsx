import { Box, Container, Typography } from '@mui/material';

export type MessageProps = {
  text: string;
};

const Message = ({ text }: MessageProps) => {
  return (
    <Container fixed>
      <Box
        fontSize={18}
        sx={{
          width: 500,
          height: 300
        }}
      >
        <Typography
          data-testid="text-error"
          marginBottom={5}
          align="center"
          fontSize={18}
        >
          <strong>{text}</strong>
        </Typography>
        <Typography fontSize={18} align="center" data-testid="text-default">
          Por favor, verifique sua pesquisa e tente novamente para obter
          resultados.
        </Typography>
      </Box>
    </Container>
  );
};

export default Message;
