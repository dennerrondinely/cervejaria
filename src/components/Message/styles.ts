import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  position: relative;
  display: flex;
  flex-direction: column;
  gap: 1rem;
`;
export const InputWrapper = styled.div`
  display: flex;
  align-items: center;
  gap: 2.2rem;
  box-shadow: 0px 2px 2px rgba(0, 0, 0, 0.25);
  border-radius: 2px;
  padding: 0.8rem 1.8rem;
  ${({ theme }) => css`
    background-color: ${theme['white']};
  `}
`;

export const Label = styled.h2`
  font-style: italic;
  font-weight: 400;
  font-size: 2rem;
  line-height: 2.3rem;
  display: flex;
  align-items: center;
  letter-spacing: 0.36px;
`;

export const Button = styled.button`
  background: transparent;
  border: none;
  cursor: pointer;
`;

export const Input = styled.input`
  width: 100%;
  font-family: 'Roboto';
  font-style: normal;
  font-weight: 400;
  font-size: 2.2rem;
  line-height: 2.6rem;
  letter-spacing: 0.36px;
  border: none;
  color: rgba(0, 0, 0, 0.3);

  &:focus-within {
    border: none;
    outline: 0;
  }
`;
