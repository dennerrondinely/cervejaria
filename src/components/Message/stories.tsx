import { Story, Meta } from '@storybook/react/types-6-0';

import Message, { MessageProps } from '.';

export default {
  title: 'Message',
  component: Message
} as Meta;

export const Default: Story<MessageProps> = (args) => <Message {...args} />;

Default.args = {
  text: 'A lista de cervejaria n√£o foi encontrada'
};
