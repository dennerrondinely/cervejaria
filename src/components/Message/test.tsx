import Message from './index';
import { render, screen } from 'utils/tests/helpers';

describe('<Message />', () => {
  it('should render the Message', () => {
    const { container } = render(<Message text="" />);

    expect(container).toContainHTML('p');
    expect(container).toContainHTML('strong');

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should to render message content', () => {
    const message = 'A lista de cervejaria n√£o foi encontrada';
    const { container } = render(<Message text={message} />);

    const textDefault = screen.getByTestId('text-default');
    expect(textDefault).toHaveTextContent(
      'Por favor, verifique sua pesquisa e tente novamente para obter resultados.'
    );

    const textError = screen.getByTestId('text-error');
    expect(textError).toHaveTextContent(message);

    expect(container.firstChild).toMatchSnapshot();
  });
});
