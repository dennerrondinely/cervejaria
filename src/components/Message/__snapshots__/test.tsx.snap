// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> should render the Message 1`] = `
.c2 {
  margin: 0;
  font-family: Quicksand;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  text-align: center;
  margin-bottom: 40px;
  font-size: 18px;
}

.c3 {
  margin: 0;
  font-family: Quicksand;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  text-align: center;
  font-size: 18px;
}

.c1 {
  font-size: 18px;
  width: 500px;
  height: 300px;
}

.c0 {
  width: 100%;
  margin-left: auto;
  box-sizing: border-box;
  margin-right: auto;
  display: block;
  padding-left: 16px;
  padding-right: 16px;
}

@media (min-width:600px) {
  .c0 {
    padding-left: 24px;
    padding-right: 24px;
  }
}

@media (min-width:600px) {
  .c0 {
    max-width: 600px;
  }
}

@media (min-width:900px) {
  .c0 {
    max-width: 900px;
  }
}

@media (min-width:1200px) {
  .c0 {
    max-width: 1200px;
  }
}

@media (min-width:1536px) {
  .c0 {
    max-width: 1536px;
  }
}

@media (min-width:1200px) {
  .c0 {
    max-width: 1200px;
  }
}

<div
  class="c0 MuiContainer-root MuiContainer-maxWidthLg MuiContainer-fixed"
>
  <div
    class="c1 MuiBox-root"
  >
    <p
      class="c2 MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter"
      data-testid="text-error"
    >
      <strong />
    </p>
    <p
      class="c3 MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter"
      data-testid="text-default"
    >
      Por favor, verifique sua pesquisa e tente novamente para obter resultados.
    </p>
  </div>
</div>
`;

exports[`<Message /> should to render message content 1`] = `
.c2 {
  margin: 0;
  font-family: Quicksand;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  text-align: center;
  margin-bottom: 40px;
  font-size: 18px;
}

.c3 {
  margin: 0;
  font-family: Quicksand;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  text-align: center;
  font-size: 18px;
}

.c1 {
  font-size: 18px;
  width: 500px;
  height: 300px;
}

.c0 {
  width: 100%;
  margin-left: auto;
  box-sizing: border-box;
  margin-right: auto;
  display: block;
  padding-left: 16px;
  padding-right: 16px;
}

@media (min-width:600px) {
  .c0 {
    padding-left: 24px;
    padding-right: 24px;
  }
}

@media (min-width:600px) {
  .c0 {
    max-width: 600px;
  }
}

@media (min-width:900px) {
  .c0 {
    max-width: 900px;
  }
}

@media (min-width:1200px) {
  .c0 {
    max-width: 1200px;
  }
}

@media (min-width:1536px) {
  .c0 {
    max-width: 1536px;
  }
}

@media (min-width:1200px) {
  .c0 {
    max-width: 1200px;
  }
}

<div
  class="c0 MuiContainer-root MuiContainer-maxWidthLg MuiContainer-fixed"
>
  <div
    class="c1 MuiBox-root"
  >
    <p
      class="c2 MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter"
      data-testid="text-error"
    >
      <strong>
        A lista de cervejaria n√£o foi encontrada
      </strong>
    </p>
    <p
      class="c3 MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter"
      data-testid="text-default"
    >
      Por favor, verifique sua pesquisa e tente novamente para obter resultados.
    </p>
  </div>
</div>
`;
